docs/architecture/assets.md

```diff
@@ -1,58 +1,81 @@
+Non-fungible assets are encoded by hashing the asset data into a `Word` and placing the `faucet_id` as the second element. Examples include NFTs like a DevCon ticket.
+
+A non-fungible asset is encoded as:  
+`[e0, faucet_id, e2, e3]`
+
+The `faucet_id` at position `1` distinguishes non-fungible from fungible assets.
 
 ### Storage
 
-[Accounts](accounts.md) and [notes](notes.md) contain asset vaults that are used to store assets. Accounts can keep unlimited assets in a sparse Merkle tree called `account vault`. Notes can store up to `255` distinct assets.
+[Accounts](accounts.md) and [notes](notes.md) have vaults used to store assets.
 
 ![Architecture core concepts](../img/architecture/asset/asset-storage.png)
 
-The information on which and how many assets are owned can be private depending on the account's or note's storage mode. This is true for any native asset in Miden.
+### Burning
+
+Assets in Miden can be burned through various methods, such as rendering them unspendable by storing them in an unconsumable note, or sending them back to their original faucet for burning using it's dedicated function.
 
-## Non-native assets
+## Alternative asset models
 
-Miden is flexible enough to create other types of assets as well.
+> All data structures not following the Miden asset model that can be exchanged.
 
-For example, developers can replicate the Ethereum ERC20 model, where ownership of fungible assets is recorded in a single account. To transact, users must send a note to that account to change the global hashmap.
+Miden is flexible enough to support other asset models. For example, developers can replicate Ethereum’s ERC20 pattern, where fungible asset ownership is recorded in a single account. To transact, users send a note to that account, triggering updates in the global hashmap state.
 
-Furthermore, a complete account can be treated as a programmable asset because ownership of accounts is transferrable. An account could be a "crypto kitty" with specific attributes and rules, and people can trade these "crypto kitties" by transferring accounts between each other.
+## Conclusion
 
-We can also think of an account representing a car. The owner of the car can change so the car account - granting access to the physical car - can be treated as an asset. In this car account, there could be rules defining who is allowed to drive the car and when.
+Miden’s asset model provides a secure, flexible, scalable, and privacy-preserving framework for representing and transferring value. By embedding asset information directly into accounts and supporting multiple asset types, Miden fosters a decentralized ecosystem where users maintain their privacy, control, transactions can scale efficiently, and censorship is minimized.
```

Comment:
I think we should include a mention of privacy somewhere in this doc page, otherwise it feels odd to conclude that assets support privacy when we haven't mentioned it before. Maybe just a sentence in the "Censorship resistance" property would be good enough.